### PR TITLE
Update tolua++ and export EffectID

### LINKED ISSUE #3505: BroadcastSoundParticleEffect doesn't work
### Expected behavior
Plugin developers can use BroadcastSoundParticleEffect in their plugins without issues.

### Actual behavior
BroadcastSoundParticleEffect does not work at the moment. The API documentation says that you should specify an effect ID, but it returns this error in the console:
```
error in function 'BroadcastSoundParticleEffect'.
     argument #2 is 'number'; 'const EffectID' expected.
```
This problem was introduced in 83870f9fc0e19da646f408a2d6636adfe4b5c002.

### Steps to reproduce the behavior
Use the super pickaxe in WorldEdit. Activate it by executing the command //, and break blocks using a pickaxe.

### PR DESCRIPTION
**Pull request by peterbell10**
This uses the new typed and scoped enum support in tolua++ to remove the "hackery" for the enum of  block ids.  Also exports `EffectID` and `SmokeDirection` which fixes #3505 .

### CODE REVIEW

**Comment by Seadragon91** [comment]
Thanks for exporting will test it.

@madmaxoft Here circleci hangs, can you take a look? Thanks :)

**Comment by Seadragon91** [comment]
Tested with world edit and works fine.

**Comment by Seadragon91** [comment]
Ah one problem I see this enums are bound to the enum class name. Is it okay that way @madmaxoft ?

**Comment by madmaxoft** [comment]
Looks good. No idea how to re-run the CircleCI build, other than @peterbell10 pushing another commit to this PR's branch

File: src/Bindings/BindingsProcessor.lua [hunk 1]
```diff
@@ -977,7 +978,38 @@ local function installEnumHook()
 				enumValues[numEnumValues] = txt
 			end
 		end
-		local res = oldEnumerate(a_Name, "{" .. table.concat(enumValues, ",") .. "}", a_VarName)
+		local res = oldEnumerate(a_Name, "{" .. table.concat(enumValues, ",") .. "}", a_VarName, a_Type)
+		res.DoxyComments = doxyComments
+		return res
+	end
+	--Hook for scoped enums
```
Comment (madmaxoft) [inline src/Bindings/BindingsProcessor.lua]: Add an empty line above here.

**Comment by madmaxoft** [comment]
@Seadragon91 what exactly do you mean by that?

**Comment by Seadragon91** [comment]
The monster type can be directly accessed by doing `mtSheep` for this ones you need to do `EffectID.SFX_RANDOM_DISPENSER_DISPENSE`

**Comment by madmaxoft** [comment]
Yes, I believe that's better

**Comment by peterbell10** [comment]
I've pushed with the added empty line but it looks like circleci still isn't working:
`Your build was not run - reason code (:github-failure)`

**Comment by madmaxoft** [comment]
This time the build was visible on CircleCI, so I restarted it and it seems to work.

**Comment by Seadragon91** [comment]
I also okay with that, but I think a example should be added. As all current global enums can be accessed directly and this ones cold confuse the dev :)

**Comment by Seadragon91** [comment]
Oh sorry then, my mistake...

**Pull request merged**